File: client/src/main/java/org/admin/utils/entities/Product.java
package org.admin.utils.entities;

import org.admin.utils.Response;

public class Product extends Response {
	private Long id;
	private String name;
	private String description;
	private Long price;
	private Integer count;
	
	public Boolean equals(Product other){
		return this.id == other.getId();
	}

	public Long getId()				{return this.id;}
	public String getName()			{return this.name;}
	public String getDescription()	{return this.description;}
	public Long getPrice()		{return this.price;}
	public Integer getCount()		{return this.count;}
	
	public void setId(Long id)						{this.id = id;}
	public void setName(String name)				{this.name = name;}
	public void setDescription(String description)	{this.description = description;}
	public void setPrice(Long price)				{this.price = price;}
	public void setCount(Integer count)				{this.count = count;}
}

File: backend/models/product.go
package models

type Product struct {
	Id    int    `json:"id" db:"id"`
	Name  string `json:"name" db:"name" binding:"required"`
	Price int    `json:"price" db:"price" binding:"required"`
	Count int    `json:"count" db:"count"`
}

type ProductUpdate struct {
	Name  *string `json:"name"`
	Price *int    `json:"price"`
	Count *int    `json:"count"`
}
